keyboards/hardwareabstraction/handwire/keymaps/default/keymap.c

```diff
@@ -0,0 +1,334 @@
+    #endif
+}
+
+void housekeeping_task_user(void){
+    #ifdef BUZZER_ENABLE
+    if(buzzer_on){
+        if(buzzer_active && timer_elapsed(buzzer_timer) > buzzer_dwell){
+        buzzer_active = false;
+        writePinLow(BUZZER_PIN);
+        }
+    }
+    #endif
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed){                         // true when pressed, false when released
+        initial_keypress = true;
+
+        #ifdef BUZZER_ENABLE
+        if(buzzer_on){
+            if(!buzzer_active){
+                buzzer_active = true;
+                buzzer_timer = timer_read();
+                writePinHigh(BUZZER_PIN);
+            }
+        }
+        #endif
+
+        switch (keycode) {
+            #ifdef HAPTIC_ENABLE
+            case KC_HPTON:
+                haptic_enable();
+                break;
+
+            case KC_HPTOFF:
+                haptic_disable();
+                break;
+
+            case KC_HPTTOG:
+                haptic_toggle();
+                break;
+
+            case KC_HPTRST:
+                haptic_reset();
+                break;
+
+            case KC_HPTFBK:
+                haptic_feedback_toggle();
+                break;
+
+            case KC_HPTCONT:
+                haptic_toggle_continuous();
+                break;
+
+            case KC_HPTDWLI:
+                haptic_dwell_increase();
+                break;
+
+            case KC_HPTDWLD:
+                haptic_dwell_decrease();
+                break;
+            #endif
+
+            #ifdef BUZZER_ENABLE
+            case KC_BZTOG:
+                buzzer_on = !buzzer_on;
+                if(buzzer_on == true){
+                    buzzer_active = true;
+                    buzzer_timer = timer_read();
+                    writePinHigh(BUZZER_PIN);
+                }
+                else{
+                    writePinLow(BUZZER_PIN);
+                }
+
+                break;
+
+            case KC_BZRST:
+                buzzer_dwell = 12;
+                break;
+
+            case KC_BZDWLI:
+                buzzer_dwell += buzzer_dwell_change;
+                break;
+
+            case KC_BZDWLD:
+                buzzer_dwell -= buzzer_dwell_change;
+                break;
+            #endif
+        }
+    }
+    return true;
+}
+
+
+//feature_oled_driver.md
+// http://javl.github.io/image2cpp/
+// Rules.mk make sure to have:
+// OLED_DRIVER_ENABLE = yes
+// WPM_ENABLE = yes
+#ifdef OLED_ENABLE
+
+#include <stdio.h>
+
+#define IDLE_FRAMES 5
+#define IDLE_SPEED 20   // Idle WPM
+
+// #define PREP_FRAMES 1
+
+#define TAP_FRAMES 2
+#define TAP_SPEED 40    // Move Bongo Cat WPM
+
+#define ANIM_FRAME_DURATION 200  // How long each frame lasts in ms
+#define ANIM_SIZE 550  // Number of bytes in array, minimize for adequate firmware size, max is 1024
+
+// Variables for display layer and haptic status on OLED
+static uint8_t layer = 0;
+char layer_str[10];
```

Comment:
```suggestion
```

Reply:
What was the reason for this suggestion? it's being used between lines 309 - 311 under oled_task_user to create a string with the current layer number to be outputted to the OLED screen.

Reply:
nevermind I see the other change you suggested.